lib/eslint.js

```diff
@@ -718,11 +720,20 @@ module.exports = (function() {
 
                 severity = getRuleSeverity(config.rules[key]);
                 options = getRuleOptions(config.rules[key]);
+                parserName = config.parser;
+
+                // for use against pre-parsed AST, may be no valid parser configured
+                try {
+                    parser = require(parserName);
```

Comment:
If we only supply `parse()`, then we would lose access to everything else on the parser, which might be useful (like `tokenize()`), so I don't think that works. 

You do bring up an interesting point, though, what is the expectation for preparsed code? I have a hard time accepting that a null parser would be expected in that case, as rules have no way of knowing that something was preparsed (nor should they). I think at a minimum we need to always provide Espree.